### PR TITLE
improvement(notification_manager): Add "Assigned to a run" notification

### LINKED ISSUE #398: People should get Email/Notification when they are assigned to jobs
right now we need to assign people, and write something for them on the discussion, to them to notice it.

It's hard to do when it's a mass of test, every assignment on a run (not a job) should send a notification/email

### PR DESCRIPTION
**Pull request by k0machi**
This commit adds both internal and email notification for assignments to
runs within argus. Manual assignments done by a user on a test run card
send a notification/email from him.

Fixes #398

### CODE REVIEW

File: argus/backend/service/client_service.py [hunk 1]
```diff
@@ -24,7 +30,24 @@ def get_model(self, run_type: str) -> PluginModelBase:
 
     def submit_run(self, run_type: str, request_data: dict) -> str:
         model = self.get_model(run_type)
-        model.submit_run(request_data=request_data)
+        run = model.submit_run(request_data=request_data)
+        if run.assignee and run.test_id:
+            self.notification_manager.send_notification(
+                receiver=run.assignee,
+                sender=g.user.id,
```
Comment (soyacz) [inline argus/backend/service/client_service.py:37]: what is g.user.id in the scope of ArgusClient?

File: argus/backend/service/client_service.py [hunk 2]
```diff
@@ -14,6 +19,7 @@ class ClientService:
     PLUGINS = {name: plugin.model for name, plugin in AVAILABLE_PLUGINS.items()}
 
     def __init__(self) -> None:
+        self.notification_manager = NotificationManagerService()
```
Comment (soyacz) [inline argus/backend/service/client_service.py:22]: why not to create it in submit_run method if needed only?

Comment (k0machi) [reply argus/backend/service/client_service.py]: Could be useful for other methods in the future.

File: argus/backend/service/client_service.py (hunk 1 above)
Comment (k0machi) [reply argus/backend/service/client_service.py]: It's the ClientService scope of argus application and `g` is the global flask context variable.

Anyway I'll remove this notification in particular, as now I think it's a bit too excessive to have a notification created on *every single* run submitted to argus.

Comment (soyacz) [reply argus/backend/service/client_service.py]: yes, might be too excessive and people will ignore it. Please remove unrelated imports then too.
I think would be good to send once a day a reminder about jobs to investigate.

Comment (k0machi) [reply argus/backend/service/client_service.py]: Done

**Review by roydahan — APPROVED** [review]
LGTM

**Review by soyacz — APPROVED** [review]
LGTM

**Review by fruch — APPROVED** [review]
I would suggest to update description and commit message that it's only doing it when update is happening

**Comment by k0machi** [comment]
> I would suggest to update description and commit message that it's only doing it when update is happening

Forgot to update the description on PR, already updated commit message.

**Pull request merged**